### PR TITLE
docs(madr): add MADR for suggesting production installation values

### PR DESCRIPTION
**Pull request by jijiechen**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues 
  - N/A
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS 
  - Confirmed
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) 
  - N/A
  - Don't forget `ci/` labels to run additional/fewer tests
    - [x] Done
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? 
  - No
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) 
  - No

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: docs/madr/decisions/046-production-installation-values.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        resource:
+          name: memory
+          target:
+            type: Utilization
+            averageUtilization: 50
+
+  podDisruptionBudget:
+    enabled: true
+
+egress:
```
Comment (lahabana) [inline docs/madr/decisions/046-production-installation-values.md:136]: I don't think egress should be part of prod recommendations

Comment (jijiechen) [reply docs/madr/decisions/046-production-installation-values.md:136]: Sure, let me remove it.

File: docs/madr/decisions/046-production-installation-values.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+```yaml
+
+controlPlane:
+  # -- Used in `zone` mode with `kdsGlobalAddress` is not empty
+  kdsGlobalAuth:
+    type: "token"
+    token: ""
+    
+  # this is a new key to restrict access to the Kuma control plane api server by cores
```
Comment (lahabana) [inline docs/madr/decisions/046-production-installation-values.md:59]: CORS ?

File: docs/madr/decisions/046-production-installation-values.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  apiServer:
+    coresAllowedDomains:
+      - "https://localhost:5681"
+      - "http://localhost:5681"
```
Comment (lahabana) [inline docs/madr/decisions/046-production-installation-values.md:63]: Should be adding whatever hostname that the API exposes.

File: docs/madr/decisions/046-production-installation-values.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+The values here will override the default `values.yaml` file, so refer to [the existing file](https://github.com/kumahq/kuma/blob/master/deployments/charts/kuma/values.yaml) if needed. 
+
+```yaml
+
+controlPlane:
+  # -- Used in `zone` mode with `kdsGlobalAddress` is not empty
+  kdsGlobalAuth:
+    type: "token"
+    token: ""
```
Comment (lahabana) [inline docs/madr/decisions/046-production-installation-values.md:57]: Can this token be provided as a k8s secret? If yes we should add it here.

File: docs/madr/decisions/046-production-installation-values.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+### Performance
+
+Improves performance by:
+- enabling HPAs for the control plane, ingress and egress components
+- setting metrics for triggering HPA to 50% CPU and memory usage to make sure pods are performing well
+
+### Proposed production installation values
+
+The values here will override the default `values.yaml` file, so refer to [the existing file](https://github.com/kumahq/kuma/blob/master/deployments/charts/kuma/values.yaml) if needed.
```
Comment (lahabana) [inline docs/madr/decisions/046-production-installation-values.md:49]: Do we need values-zone and values-global as some confs are only valid in one or the other?

File: docs/madr/decisions/046-production-installation-values.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+          name: cpu
+          target:
+            type: Utilization
+            averageUtilization: 50
+      - type: Resource
+        resource:
+          name: memory
+          target:
+            type: Utilization
+            averageUtilization: 50
```
Comment (lahabana) [inline docs/madr/decisions/046-production-installation-values.md:136]: Is this a little low? Should it be more like 80? What do other popular charts do?

File: docs/madr/decisions/046-production-installation-values.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+  podDisruptionBudget:
+    enabled: true
```
Comment (lahabana) [inline docs/madr/decisions/046-production-installation-values.md:139]: Should we disable http listener? Or at least put some comments on how to use your own certs?

File: docs/madr/decisions/046-production-installation-values.md (hunk 2 above)
Comment (jijiechen) [reply docs/madr/decisions/046-production-installation-values.md:59]: Yeah, typo.

File: docs/madr/decisions/046-production-installation-values.md (hunk 3 above)
Comment (jijiechen) [reply docs/madr/decisions/046-production-installation-values.md:63]: What are those? I assume the api server application will not limit/propose any hostname from internal. Hostnames are set by administrators, no?

File: docs/madr/decisions/046-production-installation-values.md (hunk 4 above)
Comment (jijiechen) [reply docs/madr/decisions/046-production-installation-values.md:57]: I'll delete this as we've already support this as a secret in KM.

File: docs/madr/decisions/046-production-installation-values.md (hunk 6 above)
Comment (jijiechen) [reply docs/madr/decisions/046-production-installation-values.md:136]: Let me check other implementations.

File: docs/madr/decisions/046-production-installation-values.md (hunk 7 above)
Comment (jijiechen) [reply docs/madr/decisions/046-production-installation-values.md:139]: Which http listener are you refering to?

**Comment by jijiechen** [comment]
Closing this PR as it will be tracked in a child repo

**Pull request closed without merging**